### ISSUE TITLE
[BUG]: The fasta sequence does not match the REF allele at OQ234674.1:1946

### ISSUE THREAD

**Issue by Codes1985** [CFIA-NCFAD/nf-flu#42]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Bug/Issue

Hello,

I have come across an edge case where the reference used as the top match for a sample contained a degenerate nucleotide causing the error described below.  Essentially, clair3 changed the degenerate nucleotide to "N", which resulted in an error in bcftools as the reference allele no longer matched the reference sequence.

I believe I was able to mitigate this issue making two changes:

- In the `clair3.nf`  module, I added the following flag to the `clair3` command:
     - `--keep_iupac_bases True`
     - This tells clair3 not to convert the IUPAC base to N.  This, however, resulted in a new bcftools error related to the handling of non-ATGCN characters. during the `BCF_FILTER` process.
- To compensate for this, in the `bcftools.nf`, I added the following flag to the `bcftools norm` command:
     - `-c w`
     - This corresponds to the `bcftools norm` `--check-ref` command, where the `w` sets it to "warn"; as far as I can tell, this should allow the the degenerate nucleotide to exist, but should not result in any changes to how the results are processed (see: [bcftools norm](https://samtools.github.io/bcftools/bcftools.html#norm))

In my testing, it has allowed me to successfully process the sample with the offending top match reference, though I would appreciate any feedback as to whether I have overlooked anything with my suggested parameter changes.

### Nextflow command-line

```shell
nextflow nf-flu_v3.3.2/cfia-ncfad-nf-flu-3.3.2/workflow/main.nf --input IRVC20230720IH1_Part1_Complete_AF25.csv --platform nanopore --outdir IRVC20230720IH1_Part1_Complete_AF25_nf-flu_results --major_allele_fraction 0.25 -profile singularity,slurm -resume
```

### Error Message

```shell
ERROR ~ Error executing process > 'NF_FLU:NANOPORE:BCF_CONSENSUS (RV00831-22-IAV|3_PA|OQ234674.1)'

Caused by:
  Process `NF_FLU:NANOPORE:BCF_CONSENSUS (RV00831-22-IAV|3_PA|OQ234674.1)` terminated with an error exit status (255)

Command executed:

  bgzip -c RV00831-22-IAV.Segment_3_PA.OQ234674.1.no_frameshifts.vcf > RV00831-22-IAV.Segment_3_PA.OQ234674.1.no_frameshifts.vcf.gz
  tabix RV00831-22-IAV.Segment_3_PA.OQ234674.1.no_frameshifts.vcf.gz
  
  # get low coverage depth mask BED file by filtering for regions with less than 10X
  zcat RV00831-22-IAV.Segment_3_PA.OQ234674.1.per-base.bed.gz | awk '$4<10' > low_cov.bed
  
  bcftools consensus \
    -f RV00831-22-IAV.Segment_3_PA.OQ234674.1.reference.fasta \
    -m low_cov.bed \
    RV00831-22-IAV.Segment_3_PA.OQ234674.1.no_frameshifts.vcf.gz > RV00831-22-IAV.Segment_3_PA.OQ234674.1.bcftools.consensus.fasta
  
  sed -i -E "s/^>(.*)/>RV00831-22-IAV_3_PA/g" RV00831-22-IAV.Segment_3_PA.OQ234674.1.bcftools.consensus.fasta
  
  cat <<-END_VERSIONS > versions.yml
  "NF_FLU:NANOPORE:BCF_CONSENSUS":
      bcftools: $(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*$//')
  END_VERSIONS

Command exit status:
  255

Command output:
  (empty)

Command error:
  Note: the --sample option not given, applying all records regardless of the genotype
  The fasta sequence does not match the REF allele at OQ234674.1:1946:
     REF .vcf: [TATTCAATAGCCTATATGCATCACCACAATTGGAAGGANTTTCAGCAGAGTC]
     ALT .vcf: [T]
     REF .fa : [TATTCAATAGCCTATATGCATCACCACAATTGGAAGGAYTTTCAGCAGAGTC]AAGAAAACTGCTCCTTATTGTTCAGGCTCTTAGGGACAAACTCGAACCTGGGACTTTTGATCTTGGGGGGCTATATGAAGCAATTGAGGAGTGCCTGATTAATGATCCCTGGGTTTTGCTCAATGCGTCTTGGTTCAACTCCTTCCTGACACATGCACTAAAATAGTTATAGCAGTGCTACTATTTGTTATCCGTACTGTCCAAAAAAGTA

Work dir:
/work/b6/73a925fff13d66411ea7d7776773c1

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
```

### Workflow Version

v3.3.2; revision: 91a5d05f86

### Nextflow Version

23.04.1

### Java Version

openjdk version "20-internal" 2023-03-21
OpenJDK Runtime Environment (build 20-internal-adhoc..src)
OpenJDK 64-Bit Server VM (build 20-internal-adhoc..src, mixed mode, sharing)

### Hardware

HPC Cluster

### Conda/Container Engine

None

### Additional context

[slurm-27454074_AfterClair3Change.txt](https://github.com/CFIA-NCFAD/nf-flu/files/12362487/slurm-27454074_AfterClair3Change.txt)
[slurm-27454201_OriginalError.txt](https://github.com/CFIA-NCFAD/nf-flu/files/12362488/slurm-27454201_OriginalError.txt)

**Comment by peterk87** [comment]
Thanks @Codes1985 for reporting this issue! I was hoping something like this wouldn't crop up with Clair3 and Bcftools!

I've usually gotten around this issue by changing all ambiguous bases in a ref seq to N, but I like your solution better since there's no modification and duplication of data required. I'll work this into the next patch release (3.3.3) that I'm working on in #41 since the changes are fairly minor, but critical!

**Comment by Codes1985** [comment]
You're most welcome, @peterk87!  Very happy to help.  We are rapidly approaching flu season, so it's good to iron out as many wrinkles as we can before then.  Regarding [#41](https://github.com/CFIA-NCFAD/nf-flu/pull/41), I agree.  I was going to comment there that I found an edge case where subtypes were not being reported for fluA samples because those metadata were not actually included for the sequences identified as the top BLAST match (curses, NCBI!).  So, your solution sounds great, and I think it will remedy the situation I came across, as well!
